simdistserve/base/worker.py

```diff
@@ -49,6 +49,8 @@ def __init__(
         enable_chunked_prefill=False,
         prefill_max_tokens=10 ** 7,
         decode_max_tokens=10 ** 7,
+        free_mem_slots_num=69230, # TODO: (Refactor) This is a magic number. Should be a configuration.
+        per_token_kvcache_transfertime=0.01, # TODO: (Refactor) This is a magic number. Should be a configuration.
```

Comment:
The user should specify two parameters as input: kv_cache_mem (GB) and kv_transfer_bw (Gbps) as the example [here](https://github.com/LLMServe/DistServe/tree/main/simdistserve). Use the two parameters to calculate the free_mem_slots_num and per_token_kvcache_transfertime as you like. The computation of the size of one kv_cache slot can be referred [here](https://github.com/vllm-project/vllm/blob/c83919c7a6bd47bb452321f08017ef5a5cdd553a/vllm/worker/cache_engine.py#L101) in vLLM.

After this, add another running example as follows: the user specifies the TP and PP for the prefill and decoder instance (e.g., TP=8, PP=1), you search for the optimal M:N ratio for the prefill and decoder instances. The script should output something like the example in [README](https://github.com/LLMServe/DistServe/tree/main/simdistserve):
```
python -m simdistserve.simulate_ratio \
    --prefill_tp 8 \
    --prefill_pp 1 \
    --decode_tp 8 \
    --decode pp 1 \
    --kv_cache_mem 40 \
    --kv_transfer_bw 25 \
    --model-type "facebook/opt-13b" \
    --workload sharegpt --backend distserve \
    --prefill-target 200 --decode-target 100 \
    --prefill-percentage 90 --decode-percentage 90 \
    --max-per-gpu-rate 5 \
    --esp 0.25 \
    --N 300
```
Output:
```
Best per GPU rate: 1.56
Best config: #prefill instances = M, #decoding instances = N
```